lib/ansible/modules/rpm_key.py

```diff
@@ -213,33 +231,45 @@ def execute_command(self, cmd):
             self.module.fail_json(msg=stderr)
         return stdout, stderr
 
-    def is_key_imported(self, keyid):
+    def is_key_imported(self, keyids):
         cmd = self.rpm + ' -q  gpg-pubkey'
         rc, stdout, stderr = self.module.run_command(cmd)
         if rc != 0:  # No key is installed on system
             return False
         cmd += ' --qf "%{description}" | ' + self.gpg + ' --no-tty --batch --with-colons --fixed-list-mode -'
         stdout, stderr = self.execute_command(cmd)
+
+        imported_ids = set()
         for line in stdout.splitlines():
-            if keyid in line.split(':')[4]:
-                return True
-        return False
+            imported_id = line.split(":")[4]
+            # Add imported_id to imported_ids
+            imported_ids.add(imported_id)
+
+            # Also add "short" (rpm --queryformat %{version}) keyids to imported_ids
+            for keyid in keyids:
+                # if they are "short" id (self.is_keyid(keyid)) and substring of an imported_id
+                if self.is_keyid(keyid) and keyid in imported_id:
+                    imported_ids.add(keyid)
+
+        missing_keys = set(keyids).difference(imported_ids)
+        return len(missing_keys) == 0
```

Comment:
Though, I'd probably label the expression with what it intends to compute and return that:
```suggestion
        some_gpg_keys_missing = bool(set(keyids) - imported_ids)
        return some_gpg_keys_missing
```
(I think that “return whether the length of a set of missing keys is zero” might be more difficult to parse vs. “return whether some GPG keys are missing” — it's low level vs. high level)